llvm/test/Transforms/SLPVectorizer/AArch64/reduce-fadd.ll

```diff
@@ -0,0 +1,1232 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt < %s -passes=slp-vectorizer -mtriple=aarch64-unknown-linux -S | FileCheck %s
+; RUN: opt < %s -passes=slp-vectorizer -mtriple=aarch64-unknown-linux -mcpu=neoverse-v2 \
+; RUN:       -S | FileCheck %s --check-prefix=NEOV2
```

Comment:
I think you can remove the V2 results from here now, unless you expect them to be different.

Reply:
The result is different for v16 half type.

Reply:
Different based on the cpu or based on -mattr=+fullfp16 or -mattr=+sve2?

Reply:
Different for following cases in isolation
```
-mattr=+fullfp16
-mattr=+sve2
-mcpu=neoverse-v2
```

But surprisingly same for following case !
```
-mattr=neoversev2
```

Reply:
Oh I see what you mean, the codegen is different already. That does sound odd, it is likely because the scalar cost is lower with fullfp16. And that -mcpu=neoverse-v2 implies +sve2 and +sve2 implies +fullfp16, so +fullfp16 is enough to show the difference.

Can you change the run lines to:
```
; RUN: opt < %s -passes=slp-vectorizer -mtriple=aarch64-unknown-linux -S | FileCheck %s --check-prefixes=CHECK,CHECK-FP16
; RUN: opt < %s -passes=slp-vectorizer -mtriple=aarch64-unknown-linux -mattr=+fullfp16 -S | FileCheck %s --check-prefixes=CHECK,CHECK-FP16
```

It should collapse a lot of the run lines together when they are the same. These all looks like a useful set of tests and we can hopefully get them in.

Reply:
@davemgreen The presence of function attribute as below with `-mattr=+fullfp16`
```
; NEOV2-SAME: <2 x half> [[VEC2:%.*]]) #[[ATTR0:[0-9]+]] {
```
is proving to be hindrance with merging of outputs with `check-prefixes`.

There are manual ways to get over this issue such as 
1. Checking only the required part as below with `CHECK-SAME`
```
; RUN: opt < %s -S -passes=slp-vectorizer -mtriple=aarch64-unknown-linux | FileCheck %s
; RUN: opt < %s -S -passes=slp-vectorizer -mtriple=aarch64-unknown-linux -mattr=+fullfp16 | FileCheck %s 

define half @reduce_fast_half2(<2 x half> %vec2) {
; CHECK-LABEL: define half @reduce_fast_half2(
; CHECK-SAME: <2 x half> [[VEC2:%.*]]) 
...
...
```
2. Maybe using regular expression `{{.*}}` in place of the function attribute

The cons of the manual approach here would be:
1. Auto updating tests is not possible
2. For large no of tests, this is lot of manual work

Any way to scrub the function attribute here so that using `check-prefixes` becomes possible?

Reply:
Oh I hadn't seen that before (and I got the first check prefix wrong, sorry about that).

Maybe try:
```
; RUN: opt < %s -passes=slp-vectorizer -mtriple=aarch64-unknown-linux -mattr=-fullfp16 -S | FileCheck %s --check-prefixes=CHECK,CHECK-NOFP16
; RUN: opt < %s -passes=slp-vectorizer -mtriple=aarch64-unknown-linux -mattr=+fullfp16 -S | FileCheck %s --check-prefixes=CHECK,CHECK-FP16
```
The -fullfp16 seems to work around the issue with the attributes.